### PR TITLE
Bluetooth: Mesh: BabbleSim test suite

### PR DESCRIPTION
**Pull request by trond-snekvik**
Adds a test suite for Bluetooth Mesh in bsim_bt tests, including 10
tests for common transport behavior. This test suite includes a readme
and a simple framework for adding more mesh tests in the future.

Fixes three minor issues found during testing.

### CODE REVIEW

**Comment by trond-snekvik** [comment]
cc @kruithofa

**Review by jhedberg — APPROVED** [review]

**Review by LingaoM — APPROVED** [review]

**Comment by trond-snekvik** [comment]
@nashif [This change in run_ci.sh](https://github.com/zephyrproject-rtos/zephyr/pull/32508/commits/dda6a5ee90d0f5b31e7d269761dc2ab1a2f8510c#diff-13386608ce7ff6b45dbdb3484cad93f8669ed6174c4ec7183e13db85258fa90dR25) from #32508 silently disabled all babblesim test suites in CI, killing all non-qemu CI done on the entire Bluetooth subsystem. I assume this is a reaction to #20366, but I would like to suggest that we revert this line until we can find some twister based solution to this issue.

File: tests/bluetooth/bsim_bt/bsim_test_mesh/src/mesh_test.c [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+{
+	if (bst_result != Passed) {
+		FAIL("Test timeout (not passed after %i seconds)",
+		     HW_device_time / USEC_PER_SEC);
+	}
+
+	bs_trace_silent_exit(0);
+}
+
+void bt_mesh_test_cfg(const struct bt_mesh_test_cfg *my_cfg, int wait_time)
```
Comment (aescolar) [inline tests/bluetooth/bsim_bt/bsim_test_mesh/src/mesh_test.c:173]: Minor: It's not too nice to call the function and type the same

**Review by aescolar — CHANGES_REQUESTED** [review]
Nice, thanks. (I just checked the last commit) 
Just a couple of minor issues

**Review by aescolar — APPROVED** [review]

**Pull request merged**